File: Data Structures & Algorithms/combination-target-sum/submission-9.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        res = []

        def dfs(i, curr, total):
            if total == target:
                res.append(curr.copy())
                return
            
            if i == len(nums) or total > target:
                return

            for j in range(i, len(nums)):
                # if total > target:
                #     return
                curr.append(nums[j])
                dfs(j, curr, total + nums[j])
                curr.pop()

        dfs(0, [], 0)
        return res 

File: Data Structures & Algorithms/binary-search/submission-9.cs
public class Solution {
    public int Search(int[] nums, int target) {
        int l = 0, r = nums.Length - 1;

        while( l <= r) {
            int mid = l + (r - l) / 2;
            if(target > nums[mid]) {
                l = mid + 1;
            } else if(target < nums[mid]) {
                r = mid - 1;
            } else {
                return mid;
            }
        }
        return -1;
    }
}
